From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Sun, 27 Mar 2011 15:23:55 +0200
Subject: [PATCH] autoconf cleanup

update to latest autoconf syntax

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
[osterlad: update locations for version 4.5.0]
Signed-off-by: Denis Osterland <Denis.Osterland@diehl.com>
---
 configure.ac | 116 +++++++++++++++++++++++++++++------------------------------
 1 file changed, 58 insertions(+), 58 deletions(-)

diff --git a/configure.ac b/configure.ac
index 155b34c7ecce..b87a9a44eb20 100644
--- a/configure.ac
+++ b/configure.ac
@@ -77,7 +77,7 @@ AC_ARG_ENABLE(socket-dir,
 	  esac
         ])
 	AC_MSG_RESULT(${SOCKDIR})
-	AC_DEFINE_UNQUOTED(SOCKDIR, $SOCKDIR)
+	AC_DEFINE_UNQUOTED(SOCKDIR, $SOCKDIR,[SOCKDIR])
     ]
 )
 
@@ -86,18 +86,18 @@ dnl
 dnl    ****  special unix variants  ****
 dnl
 if test -n "$ISC"; then
-  AC_DEFINE(ISC) LIBS="$LIBS -linet"
+  AC_DEFINE([ISC],[1],[ISC]) LIBS="$LIBS -linet"
 fi
 
 dnl AC_CHECKING(for OSF1)
 dnl if test -f /bin/uname ; then
 dnl if test `/bin/uname` = OSF1 || test -f /osf_boot; then
-dnl AC_DEFINE(OSF1)	# this disables MIPS again....
+dnl AC_DEFINE([OSF1],[1],[OSF1])	# this disables MIPS again....
 dnl fi
 dnl fi
 
 if test -f /sysV68 ; then
-AC_DEFINE(sysV68)
+AC_DEFINE([sysV68],[1],[sysV68])
 fi
 
 AC_CHECKING(for MIPS)
@@ -110,7 +110,7 @@ AC_TRY_LINK(,,,LIBS="$oldlibs")
 dnl
 dnl
 if test -r /dev/ptc; then
-AC_DEFINE(MIPS)
+AC_DEFINE([MIPS],[1],[MIPS])
 AC_CHECKING(wait3)
 AC_TRY_LINK(,[wait3();], ,
 AC_CHECKING(wait2)
@@ -119,7 +119,7 @@ dnl John Rouillard (rouilj@sni-usa.com):
 dnl need -I/usr/include/bsd in RISCOS otherwise sockets are broken, no
 dnl job control etc.
 dnl Detect RISCOS if wait2 is present, but not wait3.
-AC_DEFINE(USE_WAIT2) LIBS="$LIBS -lbsd" ; CC="$CC -I/usr/include/bsd"
+AC_DEFINE([USE_WAIT2],[1],[USE_WAIT2]) LIBS="$LIBS -lbsd" ; CC="$CC -I/usr/include/bsd"
 ))
 fi
 fi
@@ -136,7 +136,7 @@ if test -f /usr/lib/libpyr.a ; then
 oldlibs="$LIBS"
 LIBS="$LIBS -lpyr"
 AC_CHECKING(Pyramid OSX)
-AC_TRY_LINK(,[open_controlling_pty("")], AC_DEFINE(OSX), LIBS="$oldlibs")
+AC_TRY_LINK(,[open_controlling_pty("")], AC_DEFINE([OSX],[1],[OSX]), LIBS="$oldlibs")
 fi
 
 dnl ghazi@caip.rutgers.edu (Kaveh R. Ghazi):
@@ -161,14 +161,14 @@ main () {
 #ifdef _POSIX_VERSION
   yes;
 #endif
-], AC_NOTE(- you have a POSIX system) AC_DEFINE(POSIX) posix=1)
+], AC_NOTE(- you have a POSIX system) AC_DEFINE([POSIX],[1],[POSIX]) posix=1)
 fi
 
 AC_CHECKING(for System V)
 AC_TRY_COMPILE(
 [#include <sys/types.h>
 #include <signal.h>
-#include <fcntl.h>], [int x = SIGCHLD | FNDELAY;], , AC_DEFINE(SYSV))
+#include <fcntl.h>], [int x = SIGCHLD | FNDELAY;], , AC_DEFINE([SYSV],[1],[SYSV]))
 
 AC_CHECKING(for sequent/ptx)
 AC_EGREP_CPP(yes,
@@ -193,16 +193,16 @@ dnl (currently not used)
 dnl
 dnl AC_CHECKING(for pid_t)
 dnl AC_EGREP_CPP(pid_t,[#include <sys/types.h>
-dnl ],AC_DEFINE(PID_T_DEFINED))
+dnl ],AC_DEFINE([PID_T_DEFINED],[1],[PID_T_DEFINED]))
 dnl
 dnl AC_CHECKING(for sig_t)
 dnl AC_EGREP_CPP(sig_t,[#include <sys/types.h>
 dnl #include <signal.h>
-dnl ],AC_DEFINE(SIG_T_DEFINED))
+dnl ],AC_DEFINE([SIG_T_DEFINED],[1],[SIG_T_DEFINED]))
 dnl
 dnl AC_CHECKING(for uid_t)
 dnl AC_EGREP_CPP(uid_t,[#include <sys/types.h>
-dnl ],AC_DEFINE(UID_T_DEFINED))
+dnl ],AC_DEFINE([UID_T_DEFINED],[1],[UID_T_DEFINED]))
 dnl
 
 dnl
@@ -224,7 +224,7 @@ setpgrp();
 int y = TIOCNOTTY;
 #endif
 #endif
-], AC_NOTE(- you have jobcontrol) AC_DEFINE(BSDJOBS), AC_NOTE(- you don't have jobcontrol))
+], AC_NOTE(- you have jobcontrol) AC_DEFINE([BSDJOBS],[1],[BSDJOBS]), AC_NOTE(- you don't have jobcontrol))
 
 dnl
 dnl    ****  setresuid(), setreuid(), seteuid()  ****
@@ -236,7 +236,7 @@ setresuid(0, 0, 0);
 AC_CHECKING(setreuid)
 AC_TRY_LINK(,[
 setreuid(0, 0);
-], AC_DEFINE(HAVE_SETREUID))
+], AC_DEFINE(HAVE_SETREUID,[1],[HAVE_SETREUID]))
 dnl
 dnl seteuid() check:
 dnl   linux seteuid was broken before V1.1.11
@@ -250,7 +250,7 @@ seteuid_is_broken(0);
 #else
 seteuid(0);
 #endif
-], AC_DEFINE(HAVE_SETEUID))
+], AC_DEFINE([HAVE_SETEUID],[1],[HAVE_SETEUID]))
 
 dnl execvpe
 AC_CHECKING(execvpe)
@@ -306,23 +306,23 @@ AC_CHECKING(libtinfo)
 AC_TRY_LINK(,tgetent((char *)0, (char *)0);,,
 AC_MSG_ERROR(!!! no tgetent - no screen)))))))))
 
-AC_DEFINE(TERMINFO))
+AC_DEFINE([TERMINFO],[1],[TERMINFO])
 
 AC_CHECKING(ospeed)
-AC_TRY_LINK(extern short ospeed;,ospeed=5;,,AC_DEFINE(NEED_OSPEED))
+AC_TRY_LINK(extern short ospeed;,ospeed=5;,,AC_DEFINE([NEED_OSPEED],[1],[NEED_OSPEED]))
 
 dnl
 dnl    ****  PTY specific things  ****
 dnl
 AC_CHECKING(for /dev/ptc)
 if test -r /dev/ptc; then
-AC_DEFINE(HAVE_DEV_PTC)
+AC_DEFINE([HAVE_DEV_PTC],[1],[HAVE_DEV_PTC])
 fi
 
 AC_CHECKING(for SVR4 ptys)
 sysvr4ptys=
 if test -c /dev/ptmx ; then
-AC_TRY_LINK([],[ptsname(0);grantpt(0);unlockpt(0);],[AC_DEFINE(HAVE_SVR4_PTYS)
+AC_TRY_LINK([],[ptsname(0);grantpt(0);unlockpt(0);],[AC_DEFINE([HAVE_SVR4_PTYS],[1],[HAVE_SVR4_PTYS])
 sysvr4ptys=1])
 fi
 
@@ -331,7 +331,7 @@ AC_CHECK_FUNCS(getpt)
 dnl check for openpty()
 if test -z "$sysvr4ptys"; then
 AC_CHECK_FUNCS(openpty,,
-[AC_CHECK_LIB(util,openpty, [AC_DEFINE(HAVE_OPENPTY)] [LIBS="$LIBS -lutil"])])
+[AC_CHECK_LIB(util,openpty, [AC_DEFINE([HAVE_OPENPTY],[1],[HAVE_OPENPTY])] [LIBS="$LIBS -lutil"])])
 fi
 
 AC_CHECKING(for ptyranges)
@@ -354,8 +354,8 @@ dnl fi
 if test "$ptys" != "$pdir/pty??" ; then
 p0=`echo $ptys | tr ' ' '\012' | sed -e 's/^.*\(.\).$/\1/g' | sort -u | tr -d '\012'`
 p1=`echo $ptys | tr ' ' '\012' | sed -e 's/^.*\(.\)$/\1/g'  | sort -u | tr -d '\012'`
-AC_DEFINE_UNQUOTED(PTYRANGE0,"$p0")
-AC_DEFINE_UNQUOTED(PTYRANGE1,"$p1")
+AC_DEFINE_UNQUOTED(PTYRANGE0,"$p0",[PTYRANGE0])
+AC_DEFINE_UNQUOTED(PTYRANGE1,"$p1",[PTYRANGE1])
 fi
 
 dnl
@@ -375,7 +375,7 @@ AC_TRY_LINK([
 #define pututline _pututline
 #endif
 ],
-[int x = DEAD_PROCESS; pututline((struct utmp *)0); getutent();], AC_DEFINE(GETUTENT),
+[int x = DEAD_PROCESS; pututline((struct utmp *)0); getutent();], AC_DEFINE([GETUTENT],[1],[GETUTENT]),
 olibs="$LIBS"
 LIBS="$LIBS -lgen"
 AC_CHECKING(getutent with -lgen)
@@ -392,7 +392,7 @@ AC_TRY_LINK([
 #define pututline _pututline
 #endif
 ],
-[int x = DEAD_PROCESS; pututline((struct utmp *)0); getutent();], AC_DEFINE(GETUTENT), LIBS="$olibs")
+[int x = DEAD_PROCESS; pututline((struct utmp *)0); getutent();], AC_DEFINE([GETUTENT],[1],[GETUTENT]), LIBS="$olibs")
 )
 AC_CHECKING(ut_host)
 AC_TRY_COMPILE([
@@ -404,10 +404,10 @@ AC_TRY_COMPILE([
 #else
 #include <utmp.h>
 #endif
-],[struct utmp u; u.ut_host[0] = 0;], AC_DEFINE(UTHOST))
+],[struct utmp u; u.ut_host[0] = 0;], AC_DEFINE([UTHOST],[1],[UTHOST]))
 AC_CHECK_HEADER(utempter.h, have_utempter=yes, have_utempter=no)
 if test "$have_utempter" = yes; then
-  AC_DEFINE(HAVE_UTEMPTER)
+  AC_DEFINE([HAVE_UTEMPTER],[1],[HAVE_UTEMPTER])
   LIBS="$LIBS -lutempter"
 fi
 
@@ -420,13 +420,13 @@ test -f /usr/lib/libutil.a && LIBS="$LIBS -lutil"
 
 AC_CHECKING(getloadavg)
 AC_TRY_LINK(,[getloadavg((double *)0, 0);],
-AC_DEFINE(LOADAV_GETLOADAVG) load=1,
+AC_DEFINE([LOADAV_GETLOADAVG],[1],[LOADAV_GETLOADAVG]) load=1,
 if test -f /usr/lib/libkvm.a ; then
 olibs="$LIBS"
 LIBS="$LIBS -lkvm"
 AC_CHECKING(getloadavg with -lkvm)
 AC_TRY_LINK(,[getloadavg((double *)0, 0);],
-AC_DEFINE(LOADAV_GETLOADAVG) load=1, LIBS="$olibs")
+AC_DEFINE([LOADAV_GETLOADAVG],[1],[LOADAV_GETLOADAVG]) load=1, LIBS="$olibs")
 fi
 )
 
@@ -466,10 +466,10 @@ loadnum=3
 #endif
 ])
 
-if test -n "$load" ; then AC_DEFINE(LOADAV) fi
-if test -n "$loadtype" ; then AC_DEFINE_UNQUOTED(LOADAV_TYPE,$loadtype) fi
-if test -n "$loadnum" ; then AC_DEFINE_UNQUOTED(LOADAV_NUM,$loadnum) fi
-if test -n "$loadscale" ; then AC_DEFINE_UNQUOTED(LOADAV_SCALE,$loadscale) fi
+if test -n "$load" ; then AC_DEFINE([LOADAV],[1],[LOADAV]) fi
+if test -n "$loadtype" ; then AC_DEFINE_UNQUOTED(LOADAV_TYPE,$loadtype,[LOADAV_TYPE]) fi
+if test -n "$loadnum" ; then AC_DEFINE_UNQUOTED(LOADAV_NUM,$loadnum,[LOADAV_NUM]) fi
+if test -n "$loadscale" ; then AC_DEFINE_UNQUOTED(LOADAV_SCALE,$loadscale,[LOADAV_SCALE]) fi
 
 dnl
 dnl    ****  signal handling  ****
@@ -478,7 +478,7 @@ if test -n "$posix" ; then
 
 dnl POSIX has reliable signals with void return type.
 AC_NOTE(assuming posix signal definition)
-AC_DEFINE(SIGVOID)
+AC_DEFINE([SIGVOID],[1],[SIGVOID])
 
 else
 
@@ -489,7 +489,7 @@ AC_TRY_COMPILE(
 #ifdef signal
 #undef signal
 #endif
-extern void (*signal ()) ();], [int i;], AC_DEFINE(SIGVOID))
+extern void (*signal ()) ();], [int i;], AC_DEFINE([SIGVOID],[1],[SIGVOID]))
 AC_CHECKING(sigset)
 AC_TRY_LINK([
 #include <sys/types.h>
@@ -500,9 +500,9 @@ sigset(0, (void (*)())0);
 #else
 sigset(0, (int (*)())0);
 #endif
-], AC_DEFINE(USESIGSET))
+], AC_DEFINE([USESIGSET],[1],[USESIGSET]))
 
-AC_DEFINE(SYSVSIGS)
+AC_DEFINE([SYSVSIGS],[1],[SYSVSIGS])
 
 fi
 
@@ -528,7 +528,7 @@ AC_TRY_LINK(,[closelog();], , [oldlibs="$LIBS"
 LIBS="$LIBS -lbsd"
 AC_CHECKING(syslog in libbsd.a)
 AC_TRY_LINK(, [closelog();], AC_NOTE(- found.), [LIBS="$oldlibs"
-AC_NOTE(- bad news: syslog missing.) AC_DEFINE(NOSYSLOG)])])
+AC_NOTE(- bad news: syslog missing.) AC_DEFINE([NOSYSLOG],[1],[NOSYSLOG])])])
 
 AC_EGREP_CPP(yes,
 [#ifdef M_UNIX
@@ -548,39 +548,39 @@ AC_TRY_COMPILE([#include <sys/types.h>
 #ifdef WEXITSTATUS
   y = WEXITSTATUS(x);
 #endif
-],AC_DEFINE(BSDWAIT))
+],AC_DEFINE([BSDWAIT],[1],[BSDWAIT]))
 
 if test -z "$butterfly"; then
 AC_CHECKING(for termio or termios)
-AC_TRY_CPP([#include <termio.h>], AC_DEFINE(TERMIO),
+AC_TRY_CPP([#include <termio.h>], AC_DEFINE([TERMIO],[1],[TERMIO]),
 if test -n "$posix"; then
-AC_TRY_CPP([#include <termios.h>], AC_DEFINE(TERMIO))
+AC_TRY_CPP([#include <termios.h>], AC_DEFINE([TERMIO],[1],[TERMIO]))
 fi
 )
 fi
 
-dnl AC_CHECK_HEADER(shadow.h, AC_DEFINE(SHADOWPW))
+dnl AC_CHECK_HEADER(shadow.h, AC_DEFINE([SHADOWPW],[1],[SHADOWPW]))
 AC_CHECKING(getspnam)
-AC_TRY_LINK([#include <shadow.h>], [getspnam("x");],AC_DEFINE(SHADOWPW))
+AC_TRY_LINK([#include <shadow.h>], [getspnam("x");],AC_DEFINE([SHADOWPW],[1],[SHADOWPW]))
 
 AC_CHECKING(getttyent)
-AC_TRY_LINK(,[getttyent();], AC_DEFINE(GETTTYENT))
+AC_TRY_LINK(,[getttyent();], AC_DEFINE([GETTTYENT],[1],[GETTTYENT]))
 
 AC_CHECKING(fdwalk)
-AC_TRY_LINK([#include <stdlib.h>], [fdwalk(NULL, NULL);],AC_DEFINE(HAVE_FDWALK))
+AC_TRY_LINK([#include <stdlib.h>], [fdwalk(NULL, NULL);],AC_DEFINE([HAVE_FDWALK],[1],[HAVE_FDWALK]))
 
 AC_CHECKING(whether memcpy/memmove/bcopy handles overlapping arguments)
-AC_DEFINE(USEBCOPY)
+AC_DEFINE([USEBCOPY],[1],[USEBCOPY])
 
-AC_DEFINE(USEMEMMOVE)
+AC_DEFINE([USEMEMMOVE],[1],[USEMEMMOVE])
 
 
-AC_DEFINE(USEMEMCPY)
+AC_DEFINE([USEMEMCPY],[1],[USEMEMCPY])
 
 AC_SYS_LONG_FILE_NAMES
 
 AC_MSG_CHECKING(for vsprintf)
-AC_TRY_LINK([#include <stdarg.h>],[va_list valist; vsprintf(0,0,valist);], AC_MSG_RESULT(yes);AC_DEFINE(USEVARARGS), AC_MSG_RESULT(no))
+AC_TRY_LINK([#include <stdarg.h>],[va_list valist; vsprintf(0,0,valist);], AC_MSG_RESULT(yes);AC_DEFINE([USEVARARGS],[1],[USEVARARGS]), AC_MSG_RESULT(no))
 
 AC_HEADER_DIRENT
 
@@ -600,21 +600,21 @@ if test -z "$ac_setenv_args"; then
     ], ac_setenv_args=2)
 fi
 if test -n "$ac_setenv_args"; then
-    AC_DEFINE(USESETENV)
+    AC_DEFINE([USESETENV],[1],[USESETENV])
     if test "$ac_setenv_args" = 3; then
-        AC_DEFINE(HAVE_SETENV_3)
+        AC_DEFINE([HAVE_SETENV_3],[1],[HAVE_SETENV_3])
     elif test "$ac_setenv_args" = 2; then
-        AC_DEFINE(HAVE_SETENV_2)
+        AC_DEFINE([HAVE_SETENV_2],[1],[HAVE_SETENV_2])
     fi
 else
     AC_MSG_RESULT(no)
     AC_MSG_CHECKING(for putenv)
-    AC_TRY_LINK(,[putenv((char *)0);unsetenv((char *)0);], AC_MSG_RESULT(yes) ,  AC_MSG_RESULT(no);AC_DEFINE(NEEDPUTENV))
+    AC_TRY_LINK(,[putenv((char *)0);unsetenv((char *)0);], AC_MSG_RESULT(yes) ,  AC_MSG_RESULT(no);AC_DEFINE([NEEDPUTENV],[1],[NEEDPUTENV]))
 fi
 AC_MSG_CHECKING([for nl_langinfo(CODESET)])
 AC_TRY_LINK([
 #include <langinfo.h>
-],[nl_langinfo(CODESET);], AC_MSG_RESULT(yes);AC_DEFINE(HAVE_NL_LANGINFO), AC_MSG_RESULT(no))
+],[nl_langinfo(CODESET);], AC_MSG_RESULT(yes);AC_DEFINE([HAVE_NL_LANGINFO],[1],[HAVE_NL_LANGINFO]), AC_MSG_RESULT(no))
 
 AC_SEARCH_LIBS(gethostname, nsl)
 
@@ -629,25 +629,25 @@ if test "$enable_pam" = "yes"; then
 	pam_start(0, 0, 0, 0);
 	pam_authenticate(0, 0);
 	pam_end(0,0);
-    ], AC_MSG_RESULT(yes);AC_DEFINE(USE_PAM),
+    ], AC_MSG_RESULT(yes);AC_DEFINE([USE_PAM],[1],[USE_PAM]),
        AC_MSG_RESULT(no);LIBS="$oldlibs")
 fi
 
 AC_ARG_ENABLE(use_locale, [  --enable-use-locale         use localized month/day names])
 if test "$enable_use_locale" = "yes"; then
-  AC_DEFINE(USE_LOCALE)
+  AC_DEFINE([USE_LOCALE],[1],[USE_LOCALE])
 fi
 AC_ARG_ENABLE(telnet, [  --enable-telnet         enable builtin telnet])
 if test "$enable_telnet" = "yes"; then
-  AC_DEFINE(BUILTIN_TELNET)
+  AC_DEFINE([BUILTIN_TELNET],[1],[BUILTIN_TELNET])
 fi
 AC_ARG_ENABLE(colors256, [  --enable-colors256      enable support for 256 colors])
 if test "$enable_colors256" = "yes"; then
-  AC_DEFINE(COLORS256)
+  AC_DEFINE([COLORS256],[1],[COLORS256])
 fi
 AC_ARG_ENABLE(rxvt_osc, [  --enable-rxvt_osc       enable support for rxvt OSC codes])
 if test "$enable_rxvt_osc" = "yes"; then
-  AC_DEFINE(RXVT_OSC)
+  AC_DEFINE([RXVT_OSC],[1],[RXVT_OSC])
 fi
 
 dnl
